### PR TITLE
Implement `RefUnwindSafe` for atomic types

### LINKED ISSUE #37136: Atomic types don't implement RefUnwindSafe
They probably should.

### PR DESCRIPTION
**Pull request by apasel422**
Closes #37136

### CODE REVIEW

**Comment by alexcrichton** [comment]
@bors: r+

cc @rust-lang/libs

File: src/libstd/panic.rs [hunk 1]
```diff
@@ -231,6 +231,34 @@ impl<T: ?Sized> RefUnwindSafe for Mutex<T> {}
 #[stable(feature = "unwind_safe_lock_refs", since = "1.12.0")]
 impl<T: ?Sized> RefUnwindSafe for RwLock<T> {}
 
+#[stable(feature = "unwind_safe_atomic_refs", since = "1.14.0")]
+impl RefUnwindSafe for atomic::AtomicIsize {}
+#[stable(feature = "unwind_safe_atomic_refs", since = "1.14.0")]
```
Comment (ollie27) [inline src/libstd/panic.rs]: It doesn't make much sense for this to be marked stable when `AtomicI8` is still unstable. I think for the unstable types this should be `#[unstable(feature = "integer_atomics", issue = "32976")]` like the others impls for these types.

**Comment by apasel422** [comment]
Should the impls be placed behind `#[cfg(target_has_atomic = "x")]`?

**Comment by alexcrichton** [comment]
@bors: r-

> Should the impls be placed behind

Ah yes, indeed!

**Comment by apasel422** [comment]
As a side note, there are portions of the standard library that use atomics without the `#[cfg(target_has_atomic = "x")]` (most notably `Arc` and `sync::mpsc`). Should they be defined conditionally as well? Or does `std` assume/require their existence?

(Obviously not a change to make in this PR.)

**Comment by alexcrichton** [comment]
> As a side note, there are portions of the standard library that use atomics without the #[cfg(target_has_atomic = "x")]

True yeah, this basically just means that libstd won't compile if `AtomicUsize` isn't available (intentional).

This seems like a mildly contentious topic on the associated issue, so I'm instead going to...

@rfcbot fcp merge

**Pull request merged**